x-pack/legacy/plugins/ml/public/application/data_frame_analytics/pages/analytics_management/hooks/use_create_analytics_form/state.ts

```diff
@@ -187,3 +189,33 @@ export const getJobConfigFromFormState = (
 
   return jobConfig;
 };
+
+/**
+ * Extracts form state for a job clone from the analytics job configuration.
+ * For cloning we keep job id and destination index empty.
+ */
+export function getCloneFormStateFromJobConfig(
+  analyticsJobConfig: CloneDataFrameAnalyticsConfig
+): Partial<State['form']> {
+  const jobType: string = Object.keys(analyticsJobConfig.analysis)[0];
+
+  const resultState: Partial<State['form']> = {
+    jobType: jobType as AnalyticsJobType,
+    description: analyticsJobConfig.description ?? '',
+    sourceIndex: Array.isArray(analyticsJobConfig.source.index)
+      ? analyticsJobConfig.source.index.join(',')
+      : analyticsJobConfig.source.index,
+    modelMemoryLimit: analyticsJobConfig.model_memory_limit,
+    excludes: analyticsJobConfig.analyzed_fields.excludes,
+  };
+
+  if (jobType === JOB_TYPES.REGRESSION || jobType === JOB_TYPES.CLASSIFICATION) {
+    // @ts-ignore
+    const analysisConfig = analyticsJobConfig.analysis[jobType];
+
+    resultState.dependentVariable = analysisConfig!.dependent_variable;
+    resultState.trainingPercent = analysisConfig!.training_percent;
```

Comment:
This involved too much code to suggest in a comment but I pushed a refactor of the related types here that would solve the issue with the `// @ts-ignore` and `!`. https://github.com/elastic/kibana/commit/e476f28d33feb9b2792e28d04e143c116e951941